jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildImageSteps.java

```diff
@@ -78,11 +78,17 @@ public void run()
           ListenableFuture<Authorization> retrieveTargetRegistryCredentialsFuture =
               listeningExecutorService.submit(
                   new RetrieveRegistryCredentialsStep(
-                      buildConfiguration, buildConfiguration.getTargetRegistry()));
+                      buildConfiguration,
+                      buildConfiguration.getTargetRegistry(),
+                      buildConfiguration.getTargetImageCredentialHelperName(),
+                      buildConfiguration.getKnownTargetRegistryCredentials()));
```

Comment:
Do the target registry and cred helpers here ever come from somewhere other than the build config you pass in? Maybe you can simplify the constructor to only take a build configuration and calls `getTargetRegistry`, `getTargetImageCredentialHelperName`, etc. inside.